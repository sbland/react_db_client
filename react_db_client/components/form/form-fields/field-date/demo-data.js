export const defaultVal = new Date(Date.now());