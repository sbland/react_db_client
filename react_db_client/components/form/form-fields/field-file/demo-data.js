export const DEMO_FILES_DATA = [
    { uid: 'a', filePath: 'files/fileA.jpg', name: 'FileA' },
    { uid: 'b', filePath: 'files/fileB.jpg', name: 'FileA' },
    { uid: 'c', filePath: 'files/fileC.jpg', name: 'FileA' },
  ];
