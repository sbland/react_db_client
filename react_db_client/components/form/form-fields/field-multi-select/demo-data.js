export const demoOptions = [
    {
        uid: 'a',
        label: 'A',
    },
    {
        uid: 'b',
        label: 'B',
    },
    {
        uid: 'c',
        label: 'C',
    },
];