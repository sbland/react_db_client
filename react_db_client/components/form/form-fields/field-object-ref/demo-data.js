export const defaultVal = '123';

export const demoResults = [
  {
    _id: '123',
    uid: 'foo',
    label: 'Foo',
    name: 'FooAlt',
  },
  {
    _id: '456',
    uid: 'bar',
    label: 'Bar',
    name: 'BarAlt',
  },
];
