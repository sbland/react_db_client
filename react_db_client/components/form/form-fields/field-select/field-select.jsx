import React, { useCallback } from 'react';
import PropTypes from 'prop-types';

import { SearchAndSelectDropdown } from '@react_db_client/components.search-and-select-dropdown';
import {
  ToggleBox,
  ToggleBoxRadioGroup,
} from '@react_db_client/components.form.form-components.toggle-box';

export const FieldSelect = ({
  uid,
  unit,
  updateFormData,
  value,
  options,
  multiple,
  required,
  labelField,
  selectType,
  defaultVal,
}) => {
  const selection =
    (value && options && options.find((o) => o.uid === value)) ||
    (defaultVal && options.find((o) => o.uid === defaultVal));
  const searchFunction = useCallback(async () => options, [options]);
  switch (selectType) {
    case 'dropdown':
      return (
        <>
          <SearchAndSelectDropdown
            searchFunction={searchFunction}
            handleSelect={(newVal) => updateFormData(uid, newVal)}
            initialValue={selection}
            allowMultiple={multiple}
            searchFieldTargetField={labelField}
            labelField={labelField}
            className="formFieldInput"
            searchFieldPlaceholder={selection ? selection[labelField] : 'search...'}
            onChange={(e) => updateFormData(uid, e.target.value)}
            required={required}
            searchDelay={0}
            allowEmptySearch
          />
          {unit && <span>{unit}</span>}
        </>
      );
    case 'toggle':
      return (
        <ToggleBoxRadioGroup
          selected={options.findIndex((opt) => opt.uid === value)}
          allowDeselect
          onChange={(i, v) => updateFormData(uid, v)}
          className="flexHoriz"
        >
          {options.map((opt) => (
            <ToggleBox text={opt.label} id={opt.uid} key={opt.uid} onChange={() => {}} />
          ))}
        </ToggleBoxRadioGroup>
      );
    default:
      return <div>INVALID SELECT TYPE {selectType} </div>;
  }
};

FieldSelect.propTypes = {
  uid: PropTypes.string.isRequired,
  unit: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  updateFormData: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      uid: PropTypes.string.isRequired,
      label: PropTypes.string.isRequired,
    })
  ).isRequired,
  multiple: PropTypes.bool,
  required: PropTypes.bool,
  labelField: PropTypes.string,
  selectType: PropTypes.oneOf(['dropdown', 'toggle']),
  defaultVal: PropTypes.any,
};

FieldSelect.defaultProps = {
  unit: '',
  value: null,
  multiple: false,
  // defaultVal: null,
  required: false,
  labelField: 'label',
  selectType: 'dropdown',
  defaultVal: null,
};
