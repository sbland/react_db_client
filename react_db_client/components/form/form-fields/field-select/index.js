export { FieldSelect } from './field-select';
