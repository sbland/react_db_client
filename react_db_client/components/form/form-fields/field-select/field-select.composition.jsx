import React from 'react';
import { CompositionWrapDefault } from '@samnbuk/react_db_client.helpers.composition-wraps';
import { FieldSelect } from './field-select';
import { defaultVal, demoOptions } from './demo-data';

const updateFormData = () => {};

const defaultProps = {
  uid: 'uid',
  unit: 'unit',
  value: defaultVal,
  options: demoOptions,
  updateFormData,
};

export const BasicFieldSelect = () => (
  <CompositionWrapDefault height="4rem" width="8rem">
    <FieldSelect {...defaultProps} required />
  </CompositionWrapDefault>
);
