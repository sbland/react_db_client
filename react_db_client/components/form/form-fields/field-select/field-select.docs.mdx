---
description: 'A React Form Field Component for selection input.'
labels: ['form-field', 'field-select']
---

import { FieldSelect } from './field-select';
