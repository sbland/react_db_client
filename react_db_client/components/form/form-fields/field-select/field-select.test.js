import '@samnbuk/react_db_client.helpers.enzyme-setup';
import React from 'react';
import { mount, shallow } from 'enzyme';

import { FieldSelect } from "./field-select";
import * as compositions from './field-select.composition';
import { defaultVal, demoOptions } from './demo-data';

const updateFormData = jest.fn();

const defaultProps = {
  uid: 'uid',
  unit: 'unit',
  value: defaultVal,
  options: demoOptions,
  updateFormData,
};

describe('Field Select', () => {
    beforeEach(() => {
        updateFormData.mockClear();
      });
      test('Renders', () => {
    shallow(<FieldSelect {...defaultProps} />);
  });

  describe('Compositions', () => {
    Object.entries(compositions).forEach(([name, Composition]) => {
      test(name, () => {
        mount(<Composition />);
      });
    });
  });
  describe('shallow renders', () => {
    test('Matches Snapshot', () => {
      const component = shallow(<FieldSelect {...defaultProps} />);
      const tree = component.debug();
      expect(tree).toMatchSnapshot();
    });
  });
});
