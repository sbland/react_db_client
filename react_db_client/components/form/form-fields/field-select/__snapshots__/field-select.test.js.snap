// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field Select shallow renders Matches Snapshot 1`] = `
"<Fragment>
  <SearchAndSelectDropdown searchFunction={[Function (anonymous)]} handleSelect={[Function: handleSelect]} selectionOverride={{...}} allowMultiple={false} returnFieldOnSelect=\\"uid\\" searchFieldTargetField=\\"label\\" labelField=\\"label\\" className=\\"formFieldInput\\" searchFieldPlaceholder=\\"a\\" onChange={[Function: onChange]} required={false} searchDelay={0} allowEmptySearch={true} initialValue=\\"\\" valid={true} />
  <span>
    unit
  </span>
</Fragment>"
`;
